bat/tests/state-conversion/run.bats

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bats
+
+# shared test functions
+load ../../lib/mixerlib
+
+setup() {
+  global_setup
+
+  mixer-init-stripped-down $CLRVER 10
+  cp mixer.state current.state
+}
+
+function copy_test_state() {
+  cp $1 mixer.state
+
+  # Update test state file to current format
+  format=$(sed -n 's/[ ]*FORMAT[ ="]*\([0-9]\+\)[ "]*/\1/p' current.state)
+  sed -i "s/\(FORMAT\).*/\1 = \"$format\"/" mixer.state
+}
+
+@test "Convert all state versions to latest with version update" {
+   # Verify that all state version conversions work
+  for f in "configs"/*
+  do
+    copy_test_state $f
+
+    mixer versions update --mix-version 20
```

Comment:
For simple conversion, it would be better to use a command that does not change mixer.state, like `mixer versions --offline` for instance. Since `mixer versions update` will overwrite that value, it doesn't guarantee that the value is properly converted. It doesn't make any difference if you convert from v1.0 to v1.1 since the value is not there, but this wont be true for future versions.